itest/rust/src/engine_tests/native_structures_test.rs

```diff
@@ -95,52 +99,49 @@ fn native_structure_codegen() {
 }
 
 #[itest]
-fn native_structure_out_parameter() {
-    // Instantiate a TextServerExtension and then have Godot call a
-    // function which uses an 'out' pointer parameter.
-    let mut ext = TestTextServer::new_gd();
-    let result = ext
-        .clone()
-        .upcast::<TextServer>()
-        .shaped_text_get_carets(Rid::new(100), 200);
-
-    // Check that we called the virtual function.
-    let cell = ext.bind_mut().cell.take();
-    assert_eq!(cell, Some((Rid::new(100), 200)));
-
-    // Check the result dictionary (Godot made it out of our 'out'
-    // param).
+fn native_structure_parameter() {
+    let caret = CaretInfo {
+        leading_caret: Rect2::from_components(0.0, 0.0, 0.0, 0.0),
+        trailing_caret: Rect2::from_components(1.0, 1.0, 1.0, 1.0),
+        leading_direction: Direction::AUTO,
+        trailing_direction: Direction::LTR,
+    };
+
+    let ptr: *const _ = &caret;
+    let mut object = NativeStructTests::new_gd();
+    let result: Dictionary = object
+        .call("pass_native_struct".into(), &[ptr.to_variant()])
+        .to();
+
     assert_eq!(
-        result.get("leading_rect"),
-        Some(Variant::from(Rect2::from_components(0.0, 0.0, 0.0, 0.0)))
+        result.get("leading_caret").unwrap().to::<Rect2>(),
+        caret.leading_caret
     );
     assert_eq!(
-        result.get("trailing_rect"),
-        Some(Variant::from(Rect2::from_components(1.0, 1.0, 1.0, 1.0)))
+        result.get("trailing_caret").unwrap().to::<Rect2>(),
+        caret.trailing_caret
     );
     assert_eq!(
-        result.get("leading_direction"),
-        Some(Variant::from(Direction::AUTO))
+        result.get("leading_direction").unwrap().to::<Direction>(),
+        caret.leading_direction
     );
     assert_eq!(
-        result.get("trailing_direction"),
-        Some(Variant::from(Direction::LTR))
+        result.get("trailing_direction").unwrap().to::<Direction>(),
+        caret.trailing_direction
     );
 }
 
 #[itest]
 fn native_structure_pointer_to_array_parameter() {
-    // Instantiate a TextServerExtension.
-    let ext = TestTextServer::new_gd();
-    let result = ext
-        .clone()
-        .upcast::<TextServer>()
-        .shaped_text_get_glyphs(Rid::new(100));
+    // Instantiate a custom class.
+    let mut object = NativeStructTests::new_gd();
+    let result_ptr: *const Glyph = object.call("native_struct_array_ret".into(), &[]).to();
+    let result = unsafe { std::slice::from_raw_parts(result_ptr, 2) };
 
     // Check the result array.
     assert_eq!(result.len(), 2);
-    assert_eq!(result.at(0).get("start"), Some(Variant::from(99)));
-    assert_eq!(result.at(1).get("start"), Some(Variant::from(700)));
+    assert_eq!(result.first().unwrap().start, 99);
+    assert_eq!(result.get(1).unwrap().start, 700);
```

Comment:
```suggestion
    assert_eq!(result.at(0).start, 99);
    assert_eq!(result.at(1).start, 700);
```
No need for asymmetry

Reply:
Clippy insisted.

Reply:
Sorry, I thought this was `Array`, not a slice.

Can you not just use the `[]` operator? I can't imagine Clippy suggests replacing `[0]` with `.first().unwrap()`, that would be incredibly stupid 😬